docs/deployment.md

```diff
@@ -0,0 +1,90 @@
+# Deployments
```

Comment:
@rocketnova @lorenyu Does this type of documentation make sense here? It's moreso for adopting the template, and a project team would likely want to have more specific docs about how exactly deploys work on their project.

Reply:
I think I'd put this in [template-only-docs](https://github.com/navapbc/template-application-nextjs/tree/main/template-only-docs) maybe something like `set-up-storybook-cd.yml`. Look at these for examples:
- https://github.com/navapbc/template-infra/blob/main/template-only-docs/set-up-ci.md
- https://github.com/navapbc/template-infra/blob/main/template-only-docs/set-up-cd.md

And I'd link to it from the root README.md since app/README.md would get copied over to the project but the root README.md won't